### PR TITLE
feat(mobile):  get-authenticated-user bb-28

### PR DESCRIPTION
**Pull request by Olha-Katran**
feat(mobile): add get-authenticated-user request and save current user in app state bb-28

### CODE REVIEW

File: apps/mobile/src/slices/auth/actions.ts [hunk 1]
```diff
@@ -1,13 +1,28 @@
...
 
+const getAuthenticatedUser = createAsyncThunk<
+	UserGetAllItemResponseDto,
+	undefined,
+	AsyncThunkConfig
+>(`${sliceName}/get-authenticated-user`, async (_, { extra }) => {
+	const { authApi } = extra;
+	const response: UserGetAllItemResponseDto =
+		await authApi.getAuthenticatedUser();
+	await storage.set(StorageKey.USER, JSON.stringify(response));
```
Comment (fshabanov) [inline apps/mobile/src/slices/auth/actions.ts:21]: I don't think we need to save user in storage

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/mobile/src/slices/auth/actions.ts (hunk 1 above)
Comment (revenets) [inline apps/mobile/src/slices/auth/actions.ts:21]: I'm not sure that we actually need to store the user entity in storage.

**Review by revenets — APPROVED** [review]

File: apps/mobile/src/packages/auth/auth-api.ts [hunk 2]
```diff
@@ -20,6 +21,19 @@ class AuthApi extends BaseHttpApi {
 		super({ baseUrl, http, path: APIPath.AUTH, storage });
 	}
 
+	public async getAuthenticatedUser(): Promise<UserGetAllItemResponseDto> {
```
Comment (fshabanov) [inline apps/mobile/src/packages/auth/auth-api.ts:24]:
```suggestion
	public async getAuthenticatedUser(): Promise<UserDto> {
```

File: apps/mobile/src/slices/users/users.slice.ts [hunk 3]
```diff
@@ -8,11 +8,13 @@ import { loadAll } from "./actions";
 
 type State = {
 	dataStatus: ValueOf<typeof DataStatus>;
+	user: null | UserGetAllItemResponseDto;
```
Comment (fshabanov) [inline apps/mobile/src/slices/users/users.slice.ts:11]:
```suggestion
	user: null | UserDto;
```

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/mobile/src/packages/auth/auth-api.ts (hunk 2 above)
Comment (Olha-Katran) [reply apps/mobile/src/packages/auth/auth-api.ts]: done

File: apps/mobile/src/slices/auth/actions.ts [hunk 4]
```diff
@@ -1,21 +1,37 @@
...
 
 import { name as sliceName } from "./auth.slice";
 
+const getAuthenticatedUser = createAsyncThunk<
+	UserDto,
+	undefined,
+	AsyncThunkConfig
+>(`${sliceName}/get-authenticated-user`, async (_, { extra }) => {
+	const { authApi } = extra;
+	const response: UserDto = await authApi.getAuthenticatedUser();
```
Comment (fshabanov) [inline apps/mobile/src/slices/auth/actions.ts:19]:
```suggestion
	const user = await authApi.getAuthenticatedUser();
```
Do we need it? I think the type comes from authApi

File: apps/mobile/src/slices/auth/actions.ts [hunk 5]
```diff
@@ -1,21 +1,37 @@
...
-	UserSignUpResponseDto,
+	UserDto,
 	UserSignUpRequestDto,
 	AsyncThunkConfig
->(`${sliceName}/sign-up`, (signUpPayload, { extra }) => {
+>(`${sliceName}/sign-up`, async (signUpPayload, { extra }) => {
 	const { authApi } = extra;
 
-	return authApi.signUp(signUpPayload);
+	const response: UserSignUpResponseDto = await authApi.signUp(signUpPayload);
```
Comment (fshabanov) [inline apps/mobile/src/slices/auth/actions.ts:31]:
```suggestion
	const {user, token} = await authApi.signUp(signUpPayload);
```
Same here, do we need it?
Also use desctructuring

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/mobile/src/slices/auth/actions.ts (hunk 4 above)
Comment (Olha-Katran) [reply apps/mobile/src/slices/auth/actions.ts]: Update

File: apps/mobile/src/slices/auth/actions.ts (hunk 5 above)
Comment (Olha-Katran) [reply apps/mobile/src/slices/auth/actions.ts]: Update

**Review by fshabanov — APPROVED** [review]

**Pull request merged**